### ISSUE TITLE
Issue with VMwareHorizonClient.download.recipe - SEARCH_URL_PRODUCT_VERSION

### ISSUE THREAD

**Issue by neilmartin83** [autopkg/scriptingosx-recipes#55]
Heya,

We're seeing failures at URLTextSearcher - it looks like the URL for SEARCH_URL_PRODUCT_VERSION no longer curls down properly, even when I specify a different user agent (https://my.vmware.com/en/web/vmware/info/slug/desktop_end_user_computing/vmware_horizon_clients/5_0):

```
% curl -H "User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36" https://my.vmware.com/en/web/vmware/info/slug/desktop_end_user_computing/vmware_horizon_clients/5_0              
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>302 Found</title>
</head><body>
<h1>Found</h1>
<p>The document has moved <a href="https://my.vmware.com/en/web/vmware/downloads/info/slug/desktop_end_user_computing/vmware_horizon_clients/5_0">here</a>.</p>
</body></html>

```

Curling down https://my.vmware.com/en/web/vmware/downloads/info/slug/desktop_end_user_computing/vmware_horizon_clients/5_0 doesn't show anything useful to search for a download URL, looks like it's a rendered page so curl doesn't handle it properly.

:(

**Comment by Daz-wallace** [comment]
Still seeing an issue with this and can't seem to find away round it. The curl'd down file even with user agent headers doesn't provide any details - looks like it runs some JS to spit out the value in the GUI.

Rough intention would be to:
1) Search the URL:
https://my.vmware.com/en/web/vmware/downloads/info/slug/desktop_end_user_computing/vmware_horizon_clients/5_0
2) and find something _like_ this URL: 
https://my.vmware.com/en/web/vmware/downloads/details?downloadGroup=CART21FQ1_MAC_543&productId=863&rPId=47872
3) and search that URL to find something _like_ this:
https://download3.vmware.com/software/view/viewclients/CART21FQ1/VMware-Horizon-Client-5.4.3-16499473.dmg

**Comment by grahampugh** [comment]
I got half way there with this. The information at: https://my.vmware.com/en/web/vmware/downloads/info/slug/desktop_end_user_computing/vmware_horizon_clients/5_0 
can (in a more complicated way) also be obtained from:
https://my.vmware.com/channel/public/api/v1.0/products/getRelatedDLGList?locale=en_US&category=desktop_end_user_computing&product=vmware_horizon_clients&version=5_0&dlgType=PRODUCT_BINARY
That gives us the ability to open:
https://my.vmware.com/en/web/vmware/downloads/details?downloadGroup=CART21FQ1_MAC_543&productId=863&rPId=47872
Unfortunately, I haven't been able to find any reference to the API object that will actually give us the download link.

**Comment by grahampugh** [comment]
This should be solved by PR #58 which uses the VMware public API. 
I got a list of the endpoints from this page:
https://my.vmware.com/web/vmware/downloads/main.f5be3143ac5cd5399596.js

The solution is this:

Obtain the `downloadGroup` (`CART21FQ1_MAC_543`) and the more generic folder name (`CART21FQ1` - I've called this `clientGroup`) from this API endpoint: 
https://my.vmware.com/channel/public/api/v1.0/products/getRelatedDLGList?locale=en_US&category=desktop_end_user_computing&product=vmware_horizon_clients&version=5_0&dlgType=PRODUCT_BINARY

Use the `downloadGroup` in this API endpoint to obtain the `dmg_url`:
https://my.vmware.com/channel/public/api/v1.0/dlg/details?locale=en_US&category=desktop_end_user_computing&product=vmware_horizon_clients&version=5_0&dlgType=PRODUCT_BINARY&downloadGroup=CART21FQ1_MAC_543

Use the `clientGroup` and `dmg_url` to download the DMG:
https://download3.vmware.com/software/view/viewclients/CART21FQ1/VMware-Horizon-Client-5.4.3-16499473.dmg

**Comment by neilmartin83** [comment]
Holy moly, wow that's some serious work, thanks @grahampugh !
